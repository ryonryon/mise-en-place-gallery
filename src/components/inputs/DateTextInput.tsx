import styled from "@emotion/styled";
import * as React from "react";
import { secondary } from "../../constants/colors";

const StyledInput = styled.input`
  height: 24px;
  width: 70%;
  margin: 16px;
  padding: 4px;
  border: 1px solid ${secondary};
  color: ${secondary};
  border-radius: 4px;

  ::-webkit-datetime-edit-fields-wrapper,
  ::-webkit-datetime-edit-text,
  ::-webkit-datetime-edit-year-field,
  ::-webkit-datetime-edit-month-field,
  ::-webkit-datetime-edit-day-field,
  ::-webkit-inner-spin-button,
  ::-webkit-calendar-picker-indicator {
    color: ${secondary};
  }

  &::placeholder {
    color: ${secondary};
  }
`;

const DateTextInput = ({ ...rest }: React.HTMLAttributes<HTMLInputElement>) => <StyledInput type="date" {...rest} />;

export default DateTextInput;
