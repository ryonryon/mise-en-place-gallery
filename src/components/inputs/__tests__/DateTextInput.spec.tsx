import * as React from "react";
import * as renderer from "react-test-renderer";

import DateTextInput from "../DateTextInput";

describe("DateTextInput test", () => {
  it("tests DateTextInput renders correctly ", () => {
    const dateTextInput = renderer.create(<DateTextInput />).toJSON();
    expect(dateTextInput).toMatchSnapshot();
  });
});
