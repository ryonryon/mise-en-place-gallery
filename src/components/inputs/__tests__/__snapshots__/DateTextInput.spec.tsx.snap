// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateTextInput test tests DateTextInput renders correctly  1`] = `
<input
  className="css-19m534t"
  type="date"
/>
`;
